src/components/AppConfig/Vector.tsx

```diff
@@ -70,58 +99,67 @@ export function VectorConfig({ settings, onChange }: Props<VectorSettings>) {
-          />
-
-          <StoreConfig
-            settings={settings?.store}
-            onChange={store => onChange({ ...settings, store })}
-          />
         </>
       )}
     </FieldSet>
   )
 }
 
-function OpenAIEmbedderConfig({ settings, onChange }: Props<OpenAIEmbedderSettings>) {
-  const s = useStyles2(getStyles);
-  return (
-    <Field label="OpenAI API URL" description="" className={s.marginTop}>
-      <Input
-        width={60}
-        name="url"
-        data-testid={testIds.appConfig.openAIUrl}
-        value={settings?.url}
-        placeholder={"https://api.openai.com"}
-        onChange={e => onChange({ ...settings, url: e.currentTarget.value })}
-      />
-    </Field>
-  );
+function getOptions({ settings, secretsSet }: {
+  settings?: VectorSettings;
+  secretsSet: SecretsSet;
+}): Array<Record<string, any>> {
+  let options = [];
+  if (secretsSet.openAIKey ?? false) { // If OpenAI is configured
+    options.push({ label: "OpenAI Embeddings", value: "openai" });
+  }
+  if (settings?.store?.grafanaVectorAPI?.url !== undefined) { // If Grafana Vector API is configured
+    options.push({ label: "Grafana Vector API", value: "grafana-vector-api" });
+  }
+  return options;
 }
 
-function EmbedderConfig({ settings, onChange }: Props<EmbedderSettings>) {
+function EmbedderConfig({ settings, secrets, secretsSet, onChange }: {
+  settings?: VectorSettings;
+  secrets: Secrets;
+  secretsSet: SecretsSet;
+  onChange: (settings: EmbedderSettings) => void;
+}) {
+  let options = getOptions({settings, secretsSet});
+  let placeholder = options.length > 0 ? "Select an option" : "Please configure OpenAI or Grafana Vector API above";
   return (
     <>
       <h4>Embedder</h4>
-      <Field label="Embedder Type" description="The type of embedder">
+      <Field label="Embedder Option" description="Embedder setup option">
         <Select
-          options={[
-            { label: "OpenAI", value: "openai" },
-          ]}
-          value={settings?.type}
-          onChange={e => e.value !== undefined && onChange({ ...settings, type: e.value })}
+          options={options}
+          placeholder={placeholder}
+          value={settings?.embed?.type}
+          onChange={e => {
+            if (e.value !== null) {
+              let newSettings: EmbedderSettings = { ...settings, type: e.value };
+          
+              if (e.value === "OpenAI") {
+                newSettings.authType = "token-auth";
+                newSettings.authSettings = {
+                  secureJsonData: {
+                    authToken: secrets.openAIKey!,
+                  },
+                  secureJsonFields: {
+                    authToken: secretsSet.openAIKey!,
+                  },
+                };
+              } else if (e.value === "VectorAPI") {
+                newSettings.authType = settings?.store?.authType;
+                newSettings.authSettings = settings?.store?.authSettings;
+              }
```

Comment:
Mentioned in another comment but e.g. here, `authSettings` shouldn't exist really, instead we should just store the inner fields on the `secrets` object. See the `openAIKey` secret field for an example!